frontend/src/metabase/lib/expressions/typechecker.js

```diff
@@ -64,7 +64,7 @@ export function typeCheck(cst, rootType) {
       const clause = CLAUSE_TOKENS.get(functionToken);
       const name = functionToken.name;
       const expectedArgsLength = clause.args.length;
-      if (clause.args.length !== args.length) {
+      if (!clause.multiple && clause.args.length !== args.length) {
```

Comment:
For variadic functions should we check that `clause.args.length <= args.length`? I'm not 100% sure that all the clause args are correct though, so maybe better to leave it unchecked and avoid false errors.

Reply:
The configs for the variadic functions are mostly wrong :-) That's why I did the above, just minimalistic for now, until I have a chance to cleanup the config (and tackle the matching of argument types, e.g. number vs string).